### PR TITLE
feat: add more unit tests for Json writing

### PR DESCRIPTION
**Pull request by asalzburger**
This PR adds more testing to the Json converters:
- the most common grid types are now tested for writing reading
- the detector volume finder is also written as such

### CODE REVIEW

File: Tests/UnitTests/Plugins/Json/DetectorVolumeFinderJsonConverterTests.cpp [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+  auto readInRootVolumeFinder =
+      Acts::DetectorVolumeFinderJsonConverter::fromJson(rFinderJson);
+
+  BOOST_CHECK(readInRootVolumeFinder.instance() != nullptr);
+
+  auto readInIndexedDetectorVolumesImpl =
+      dynamic_cast<const IndexedDetectorVolumesImpl*>(
+          readInRootVolumeFinder.instance());
+
+  BOOST_CHECK(readInIndexedDetectorVolumesImpl != nullptr);
```
Comment (AJPfleger) [inline Tests/UnitTests/Plugins/Json/DetectorVolumeFinderJsonConverterTests.cpp:82]:
```suggestion
  BOOST_REQUIRE(readInRootVolumeFinder.instance() != nullptr);

  auto readInIndexedDetectorVolumesImpl =
      dynamic_cast<const IndexedDetectorVolumesImpl*>(
          readInRootVolumeFinder.instance());

  BOOST_REQUIRE(readInIndexedDetectorVolumesImpl != nullptr);
```
We now use `<level>=REQUIRE` for the `nullptr` checks, when we try to access a member later.
Difference is:
`CHECK` just collects the failures
`REQUIRE` stops the test immediately (safer for the above case)

File: Tests/UnitTests/Plugins/Json/GridJsonConverterTests.cpp [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+  using EqBoundEqClosed =
+      Acts::Experimental::detail::GridAxisGenerators::EqBoundEqClosed;
+
+  EqBoundEqClosed eqBoundEqClosed{{-6., 6.}, 3, {-M_PI, M_PI}, 3};
+  // Create the grid with the provided axis generator
+  using GridTypeEQBEQC =
+      typename EqBoundEqClosed::template grid_type<std::size_t>;
+  GridTypeEQBEQC eqBoundEqClosedGrid(eqBoundEqClosed());
+
+  // Lets write in local coordinates
```
Comment (AJPfleger) [inline Tests/UnitTests/Plugins/Json/GridJsonConverterTests.cpp:207]:
```suggestion
  // Let's write in local coordinates
```

File: Tests/UnitTests/Plugins/Json/GridJsonConverterTests.cpp [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+BOOST_AUTO_TEST_CASE(Grid2DSingleEntryBound) {
+  using EqBoundEqBound =
+      Acts::Experimental::detail::GridAxisGenerators::EqBoundEqBound;
+
+  EqBoundEqBound eqBound2{{0., 5.}, 5, {0., 2.}, 2};
+  // Create the grid with the provided axis generator
+  using GridTypeEQB2 = typename EqBoundEqBound::template grid_type<std::size_t>;
+  GridTypeEQB2 eqBound2Grid(eqBound2());
+
+  // Lets write in local coordinates
```
Comment (AJPfleger) [inline Tests/UnitTests/Plugins/Json/GridJsonConverterTests.cpp:152]:
```suggestion
  // Let's write in local coordinates
```

**Review by AJPfleger — CHANGES_REQUESTED** [review]

**Review by AJPfleger — APPROVED** [review]

**Comment by acts-project-service** [comment]
## :red_circle: Athena integration test results [27fc8ec2a7e182649203c08faa901f180b3e6f86]
### :red_circle: Some tests have failed!
Please investigate the [pipeline](https://gitlab.cern.ch/acts/acts-athena-ci/-/pipelines/6499943)!

|status|job|
|------|---|
|:green_circle: | [run_unit_tests](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34065871) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsBenchmarkWithSpot.sh 8 100](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34065870) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsWorkflow.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34065869) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateAmbiguityResolution.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34065868) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateResolvedTracks.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34065867) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateTracks.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34065866) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateActsCoreSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34065865) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateActsSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34065864) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34065863) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateOrthogonalSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34065862) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateClusters.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34065861) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsPersistifyEDM.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34065860) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsKfRefitting.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34065859) |
|:green_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsExtrapolationAlgTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34065858) |
|:green_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsITkTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34065857) |
|:green_circle: | [run_workflow_tests_run4_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34065856) |
|:red_circle: | [run_workflow_tests_run2_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34065855) |
|:red_circle: | [run_workflow_tests_run3_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34065854) |
|:red_circle: | [run_workflow_tests_run3_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34065853) |

**Pull request merged**